src/saturation_adjustment.jl

```diff
@@ -93,59 +45,23 @@ See also [`saturation_adjustment`](@ref).
     _T_min = TP.T_min(param_set)
     tol = RS.RelativeSolutionTolerance(relative_temperature_tol)
 
-    T_1 = max(_T_min, air_temperature(param_set, e_int, PhasePartition(q_tot))) # Assume all vapor
-    if T_1 ≥ _T_min
-        q_v_sat = q_vap_saturation(param_set, T_1, ρ, phase_type)
-        unsaturated = q_tot <= q_v_sat
-        if unsaturated
-            return T_1
-        end
-    end
-    _T_freeze = TP.T_freeze(param_set)
-    @inline e_int_sat(T) =
-        internal_energy_sat(param_set, ReLU(T), ρ, q_tot, phase_type)
-    temperature_tol = _T_freeze * relative_temperature_tol
-    e_int_upper = e_int_sat(_T_freeze + temperature_tol / 2) # /2 => resulting interval is `temperature_tol` wide
-    if e_int < e_int_upper
-        e_int_lower = e_int_sat(_T_freeze - temperature_tol / 2) # /2 => resulting interval is `temperature_tol` wide
-        if e_int_lower < e_int # < e_int_upper
-            return _T_freeze
-        end
+    # Unsaturated Check
+    T_1 = max(_T_min, air_temperature(param_set, e_int, PhasePartition(q_tot)))
+    if T_1 >= _T_min && q_tot <= q_vap_saturation(param_set, T_1, ρ, phase_type)
```

Comment:
Isn't T_1 >= _T_min always true as we do T_1 = max(_T_min, ...)?

Reply:
Right, only the q_tot < q_sat condition really matters here. We can remove the temperature check.

Reply:
There are some other places in the PR where we can remove the check too.

Reply:
Will do